vm.go

```diff
@@ -186,6 +187,41 @@ func (vm *VM) evaluateSnippet(filename string, snippet string, kind evalKind) (o
 	return output, nil
 }
 
+func (vm *VM) evaluateDependencies(fileName string, children []ast.Node, dependencies *[]string, stackTrace *[]traceFrame, recursionDepth int) (err error) {
+	if recursionDepth > vm.MaxStack {
```

Comment:
I think we can drop this check. This is only helpful to prevent infinite loop in case of recursive imports, but there is a better way. We can just check if we visited an import already and skip it (it's enough to process each import only once). What we're doing here is basically DFS.

This will simplify things a bit – we no longer need to care about stackTrace at all.

Reply:
Ah. I hadn't thought about infinite loops. I'll fix it.

Reply:
Fixed.